sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3545,6 +3556,19 @@ ur_result_t ExecCGCommand::enqueueImpQueue() {
     const AdapterPtr &Adapter = MQueue->getAdapter();
     if (MEvent != nullptr)
       MEvent->setHostEnqueueTime();
+    // Honor explicit dependencies set with depends_on.
+    // Important for cross queue dependencies.
+    // Adding EventsWait output event to the barrier wait list to build proper
+    // dependency graph
+    if (RawEvents.size()) {
+      ur_event_handle_t Event{};
+      auto Result = Adapter->call_nocheck<UrApiKind::urEnqueueEventsWait>(
+          MQueue->getHandleRef(), RawEvents.size(),
+          RawEvents.size() ? &RawEvents[0] : nullptr, &Event);
```

Comment:
Same as above, `.size()` is known to be non-zero.

Reply:
https://github.com/intel/llvm/pull/18139/commits/c64a71a6df9418858f95d604982728ed6e301136